MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2541,33 +2563,58 @@ public string GetHardwareAcceleratedVideoDecoder(VideoType videoType, MediaStrea
                         case "h264":
                             if (_mediaEncoder.SupportsDecoder("h264_qsv") && encodingOptions.HardwareDecodingCodecs.Contains("h264", StringComparer.OrdinalIgnoreCase))
                             {
-                                // qsv decoder does not support 10-bit input
-                                if ((videoStream.BitDepth ?? 8) > 8)
-                                {
-                                    encodingOptions.HardwareDecodingCodecs = Array.Empty<string>();
-                                    return null;
-                                }
-                                return "-c:v h264_qsv ";
+                                return "-c:v h264_qsv";
                             }
                             break;
                         case "hevc":
                         case "h265":
                             if (_mediaEncoder.SupportsDecoder("hevc_qsv") && encodingOptions.HardwareDecodingCodecs.Contains("hevc", StringComparer.OrdinalIgnoreCase))
                             {
-                                //return "-c:v hevc_qsv -load_plugin hevc_hw ";
-                                return "-c:v hevc_qsv ";
+                                if (isColorDepth10)
+                                {
+                                    if (encodingOptions.EnableDecodingColorDepth10)
+                                    {
+                                        return "-c:v hevc_qsv";
+                                    }
+
+                                    return null;
+                                }
+
+                                return "-c:v hevc_qsv";
```

Comment:
```suggestion
                                return (isColorDepth10 && !encodingOptions.EnableDecodingColorDepth10) ? null : "-c:v hevc_qsv";
```
I think this is a good place to use a ternary